lib/zendesk_api/middleware/request/retry.rb

```diff
@@ -8,19 +7,37 @@ module Request
       # @private
       class Retry < Faraday::Middleware
         DEFAULT_RETRY_AFTER = 10
-        ERROR_CODES = [429, 503]
+        DEFAULT_ERROR_CODES = [429, 503]
 
         def initialize(app, options = {})
           super(app)
           @logger = options[:logger]
+          @error_codes = options.key?(:retry_codes) && options[:retry_codes] ? options[:retry_codes] : DEFAULT_ERROR_CODES
+          @retry_on_exception = options.key?(:retry_on_exception) && options[:retry_on_exception] ? options[:retry_on_exception] : false
         end
 
         def call(env)
           original_env = env.dup
-          response = @app.call(env)
+          exception_happened = false
+          if @retry_on_exception
+            begin
+              response = @app.call(env)
+            rescue StandardError => e
+              exception_happened = true
+            end
+          else
+            response = @app.call(env)
+          end
+
+          if exception_happened || @error_codes.include?(response.env[:status])
+
+            if exception_happened
+              seconds_left = DEFAULT_RETRY_AFTER.to_i
+              @logger.warn "An exception happened, waiting #{seconds_left} seconds... #{e}" if @logger
+            else
+              seconds_left = (response.env[:response_headers][:retry_after] || DEFAULT_RETRY_AFTER).to_i
+            end
 
-          if ERROR_CODES.include?(response.env[:status])
-            seconds_left = (response.env[:response_headers][:retry_after] || DEFAULT_RETRY_AFTER).to_i
             @logger.warn "You have been rate limited. Retrying in #{seconds_left} seconds..." if @logger
```

Comment:
With the more expansive list of conditions for which we're retrying, this logging is too specific.  Not like we're going to pay attention to it every time anyway.

Reply:
I think this was there before (and has assertions it happens!)